### DISCUSSION TITLE
J 4.x – Module for single contact of com_contact

### DISCUSSION THREAD

**Discussion by josarnold** [joomla/joomla-cms#47093] [2. Cold Features]
Since the Joomla build in contact component is fully equipped through the field functionality, it is most useful for all kinds of forms. Unfortunately a module for com_contact is missing. 

It would be great to have a module for a single contact form of com_contact in the core.

**Comment by brianteeman** [comment]
agree

**Comment by viocassel** [comment]
+1

**Comment by chmst** [comment]
It is rather simple to write such a module and I suppose that every developer has own modules for custom projects. At least I have but I know that then will follow lots of requirements and people will not be content with a simple solution.

**Comment by alikon** [comment]
there are already 3dp on JED for this i think

**Comment by brianteeman** [comment]
> there are already 3dp on JED for this i think

that use com_contact data ?

**Comment by josarnold** [comment]
> there are already 3dp on JED for this i think

> that use com_contact data ?

No, certainly not.

**Comment by josarnold** [comment]
> It is rather simple to write such a module and I suppose that every developer has own modules for custom projects. At least I have but I know that then will follow lots of requirements and people will not be content with a simple solution.

I am excited to hear, that the solution would be simple. Despite of further ideas we don't know yet – so let's just do it.

**Comment by josarnold** [comment]
I still think that's a good idea and end users would really appreciate it very much.

**Comment by alikon** [comment]
this is what i've done for a friend of mine https://github.com/alikon/testcom/tree/main/modules/mod_contact

**Comment by josarnold** [comment]
Thank You very much. This gives a category-list of contacts, right? – I'll check it out. Maybe it comes handy for my own ragged attempts I am struggling with, to get a single contact-form in a module .

**Comment by brianteeman** [comment]
> this is what i've done for a friend of mine https://github.com/alikon/testcom/tree/main/modules/mod_contact

That produces a list of contacts in a module. The feature request is for a single contact form module using a contact defined in the contact component.



> It is rather simple to write such a module and I suppose that every developer has own modules for custom projects. At least I have but I know that then will follow lots of requirements and people will not be content with a simple solution.

Can you share what you have done. Maybe its "good enough" or someone will find the time to "enhance it"

**Comment by joomlafun** [comment]
If a module there is, options to :
-have the front-end custom field placed at the top of the form
(for now I use the css flex order property to solves this)
-have the ability to remove the mandatory "subject" field 
Would be great.

By the way, a lot of people are purchasing contact form extensions when they don't have needs for it and I think this is because com_contact is not the most easy component to use in Joomla.
For example  its front-end custom field configuration which is mainly what users want to do with a contact form could be more reachable.

Adding a module will add reachability for joomla beginners, but maybe the ease of use of the component could at the same time be reviewed for the custom field part.

Of course people can read documentation but this tool is not far from being super great and reach plenty of people saving them money.
